subscriptions/tasks.py

```diff
@@ -595,3 +678,25 @@ def run(self):
 
 
 fire_week_estimate_last = FireWeekEstimateLast()
+
+
+class RequeueFailedTasks(Task):
+
+    """
+    Task to requeue failed schedules.
+    """
+    name = "subscriptions.tasks.requeue_failed_tasks"
+
+    def run(self, **kwargs):
+        l = self.get_logger(**kwargs)
+        failures = SubscriptionSendFailure.objects.all()
+        l.info("Attempting to requeue <%s> failed Subscription sends" %
+               failures.count())
+        for failure in failures:
+            subscription_id = str(failure.subscription_id)
+            # Cleanup the failure before requeueing it.
+            failure.delete()
+            send_next_message.delay(subscription_id)
```

Comment:
We are now catching the HTTP errors in the send_message task and setting the status back to zero before retrying or hitting the celery max retries, so we shouldn't get any left in status = 1.